walk-in-interview/src/main/webapp/script.js

```diff
@@ -122,98 +109,205 @@ function renderSelectOptions(id) {
  */
-async function renderJobListings(sortBy, order, pageSize, pageIndex) {
-  const jobListings = await getJobListings(sortBy, order, pageSize, pageIndex)
-      .catch((error) => {
-        console.log('error', error);
-        setErrorMessage(/* msg */ RESPONSE_ERROR,
-            /** include default msg */ false);
-      });
+function displayJobListings(jobPageData) {
   const jobListingsElement = document.getElementById('job-listings');
+  const jobShowing = document.getElementById('job-listings-showing');
 
-  /** reset the list so we don't render the same jobs twice */
+  /* reset the list so we don't render the same jobs twice */
   jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
   const jobListingTemplate = document.getElementById('job-listing-template');
 
   jobListings.forEach((job) => {
     const jobListing =
-      jobListingTemplate.cloneNode( /** and child elements */ true);
+      jobListingTemplate.cloneNode( /* and child elements */ true);
     jobListing.setAttribute('id', job['jobId']);
 
     const jobTitle = jobListing.children[0];
     jobTitle.innerText = job['jobTitle'];
 
     const jobAddress = jobListing.children[1];
-
     const location = job['jobLocation'];
     jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
 
+    const jobPay = jobListing.children[2];
+    const pay = job['jobPay'];
+    jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
+      `(${pay['paymentFrequency'].toLowerCase()})`;
+
+    const requirementsList = jobListing.children[3];
+    const fullRequirementsList = getRequirementsList();
+    const requirementsArr = [];
+
+    job['requirements'].forEach((req) => {
+      requirementsArr.push(fullRequirementsList[req]);
+    });
+
+    requirementsList.innerText =
+      `Requirements List: ${requirementsArr.join(', ')}`;
+
     jobListingsElement.appendChild(jobListing);
   });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Add the list of jobs that are stored in the database
+ * given the filter fields.
+ */
+async function renderJobListings() {
+  if (!validateFilters()) {
+    return;
+  }
+
+  const sortingParam = document.getElementById('sort-by').value;
+
+  /*
+   * Note that this platform currently only sorts by Salary.
+   * TODO(issue/62): support more filters
+   */
+  if (!sortingParam.includes(SALARY_PARAM)) {
+    console.error('this app currently only sorts by salary');
+    return;
+  }
+
+  const sortByParam = SALARY_PARAM;
```

Comment:
Maybe we can get these param out from the document element here, then pass these param to the validateFilterParams(param1, param2...) 
then call
getJobListing(param1, param2...)
This way we're sure that everyone is looking at the same param?